command/attach_artifacts_test.go

```diff
@@ -107,6 +107,13 @@ func (s *ArtifactsSuite) TestArtifactErrorsIfDoesNotExist() {
 	s.Len(s.cmd.Files, 0)
 }
 
+func (s *ArtifactsSuite) TestArtifactSkipsErrorWithOptionalArgument() {
+	s.cmd.Files = []string{"foo"}
+	s.cmd.Optional = true
+	s.Error(s.cmd.Execute(s.ctx, s.comm, s.logger, s.conf))
```

Comment:
Shouldn't this not error?

Reply:
It errors because the buildFileList walks the tree and only returns files that actualy exist that match the pattern.